stonesoup/updater/categorical.py

```diff
@@ -40,96 +96,35 @@ def _check_measurement_model(self, measurement_model):
             else:
                 measurement_model = self.measurement_model
 
-        try:
-            measurement_model.emission_matrix
-        except AttributeError:
-            raise ValueError("Measurement model must be categorical. I.E. it must have an "
-                             "Emission matrix property for the HMMUpdater")
+        if not isinstance(measurement_model, MarkovianMeasurementModel):
+            raise ValueError(
+                "HMMUpdater must be used in conjuction with HiddenMarkovianMeasurementModel types"
+            )
 
         return measurement_model
 
-    def _get_emission_matrix(self, hypothesis):
-
-        measurement_model = self._check_measurement_model(hypothesis.measurement.measurement_model)
-
-        return measurement_model.emission_matrix
-
-    def predict_measurement(self, predicted_state, measurement_model=None, category_names=None,
-                            **kwargs):
-        r"""Predict the measurement implied by the predicted state mean
+    def predict_measurement(self, predicted_state, measurement_model, measurement, **kwargs):
```

Comment:
Why is the `measurement` needed here? Are their attributes which should be part of the model?

Reply:
It was passed here so that the measurement category names could be copied across to the predicted measurement.
Currently, measurement category names are added on by the HMMSensor, NOT by the MarkovianMeasurementModel.
Do you think the measurement model should contain these instead?

Reply:
Have moved the `measurement_categories` to the measurement model. The HMMSensor now grabs this attribute and adds it to the yielded CategoricalDetections.